restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1710,15 +1750,35 @@ private void outbound(final Object message, final ActorRef sender) throws Servle
         ec.executePreOutboundAction(request, this.extensions);
         switch (request.type()) {
             case CLIENT: {
-                if (request.isAllowed()) {
-                    outboundToClient(request, sender);
+                ClientsDao clients = storage.getClientsDao();
+                Client client = clients.getClient(request.to().replaceFirst("client:", ""));
+                if (client != null) {
+                    long delay = sendPushNotificationIfNeeded(client.getPushClientIdentity());
```

Comment:
@agafox Similar to the P2P calls, the Push Notification request should be sent AFTER the extension has been executed and when the call is allowed, that is after `if (request.isAllowed()) {`